### ISSUE TITLE
nfs plugin: Unexpected number of fields for NFSv4 server statistics: 62

### ISSUE THREAD

**Issue by didib** [collectd/collectd#2076]
*   Version of collectd:
collectd-5.5.2-2.el7.x86_64
*   Operating system / distribution:
CentOS 7
# uname -r
3.10.0-327.36.3.el7.x86_64

## Expected behavior
No error

## Actual behavior
nfs plugin: Unexpected number of fields for NFSv4 server statistics: 62

## Steps to reproduce
Not sure.
I use collectd.conf which is a variation on [1].
I get this error on every (re)start.

[1] https://github.com/sradco/ovirt-metrics-conf

**Comment by rubenk** [comment]
This looks like a duplicate of #1304

**Comment by didib** [comment]
I considered adding this as a comment to #1304, then decided it's not exactly the same (client vs server) and would rather not spam that one. NP closing as duplicate, as long as it's not forgotten.

**Comment by rubenk** [comment]
No you're right, I missed that this was about the server, not the client.

**Comment by ConiKost** [comment]
Any news on this? I am also hitting that message :/

Jul 17 11:17:00 [collectd] nfs plugin: Unexpected number of fields for NFSv4 server statistics: 62.

**Comment by shastah** [comment]
That's because in `src/nfs.c`, in `nfs_read_linux()` you have:
```
  char *fields[64];
```
That's not enough. On my 4.13.12 kernel with `CONFIG_NFSD_V4`:
```
$ awk '/^proc/ {print $1, $2}' /proc/net/rpc/nfsd
proc3 22
proc4 2
proc4ops 72    <-------
$
```
so `fields` should be bumped to at least 74 (two first columns plus 72 fields). But I don't understand why is this hardcoded, instead of declaring it ReallyBig(TM) and then only using first 2 + $whatever_second_column_says. Or using `MAX(NFS4_SERVER_MAX_PROC, NFS4_CLIENT_MAX_PROC)` (as `nfs_read_linux()` is used to read both server and client stats)

**Comment by thlapin** [comment]
We're experiencing this since moving from 5.7 to 5.8. 
On CentOS 7 with nfs-utils-1.3.0-0.48.el7_4.x86_64.

**Comment by sradco** [comment]
We have the same issue.
CentOS 7.4 , collectd 5.8.0.

**Comment by emanuel-b-q** [comment]
We experience this also with RHEL 7.3 and collectd-5.8.0-1.el7.x86_64

**Comment by iitx24** [comment]
I have same issue with Redhat 7.2.

**Comment by blackknight36** [comment]
Having the same issue on CentOS 7.4.  Is there any way to fix this?

**Comment by blackknight36** [comment]
As a workaround I have created a small script that works with the exec plugin to collect NFS 4 metrics.

https://gist.github.com/blackknight36/9c8b29eb3c0f2bdbb339239d5da5fa6d#file-nfs4_stats-rb

This only collects server stats and only works with NFS v4.  Support for client stats and nfs v3 could easily be added if needed.

**Comment by rpv-tomsk** [comment]
Closed in #2915 .
